File: 3158-find-the-xor-of-numbers-which-appear-twice/3158-find-the-xor-of-numbers-which-appear-twice.py
class Solution:
    def duplicateNumbersXOR(self, nums: List[int]) -> int:
        count = {}

        # Count occurence of each number
        for num in nums:
            if num in count:
                count[num] +=1
            else:
                count[num] = 1
        
        # Find numbers that appear twice and xor them
        xor_result = 0
        for num, times in count.items():
            if times == 2:
                xor_result ^= num
        
        return xor_result

File: 3158-find-the-xor-of-numbers-which-appear-twice/3158-find-the-xor-of-numbers-which-appear-twice.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var duplicateNumbersXOR = function (nums) {
    seen = new Set();
    let xor_result = 0;

    for (let num of nums) {
        if (seen.has(num)) {
            xor_result ^= num;
        }
        else {
            seen.add(num);
        }
    }
    return xor_result;
};